main.rb

```diff
@@ -0,0 +1,23 @@
+def bubble_sort(arr)
+  number_ofloop = arr.length - 1
+  (0..number_ofloop).each do |_i|
+    (0...number_ofloop).each do |j|
+      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
+    end
+  end
+  arr
+end
+
+puts bubble_sort([4, 3, 78, 2, 0, 2])
+
+# The second:
+
+def bubble_sort_by(arr)
+  number_ofloop = arr.length - 1 # loop inside the index not the length
+  (0..number_ofloop).each do |_i|
```

Comment:
- [ ] **OPTIONAL** It is a convention in Ruby to use the only `_` for unused parameters, or you could skip the parameters as well and leave it like: `(0..number_ofloop).each do`